prql-compiler/src/sql/gen_expr.rs

```diff
@@ -184,6 +183,60 @@ pub(super) fn translate_literal(l: Literal, ctx: &Context) -> Result<sql_ast::Ex
     })
 }
 
+fn translate_datetime_literal(
```

Comment:
@aljazerzen this is an example of the question in https://github.com/PRQL/prql/issues/873 — should this be a method on `DialectHandler`, rather than a function which branches based on the value of `ctx.dialect`?

Reply:
Yes, that would be object-oriented-approach best practice. That does not apply in Rust - I dont't think it matters where we implement this complexity, so I can remain here for now.